src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_modular.py

```diff
@@ -3254,20 +3261,13 @@ def prepare_control_image(
 
         return image
 
-    # Copied from diffusers.pipelines.stable_diffusion_xl.pipeline_stable_diffusion_xl.StableDiffusionXLPipeline.encode_prompt
```

Comment:
I was thinking that we could have a new `_encode_prompt` method ( or maybe better to have a public method for this, so use a different name),  and refactor the current `encode_prompt` to use that new method,  what do you think?

Reply:
Yeah that works and I agree a public method is better for this, how about `encode_single_prompt`?

Reply:
sounds good! and we can discuss with the team and change it later (it will be under this PR for now but I think we should change that for regular pipeline too)

Reply:
I've added `encode_single_prompt` and `encode_prompt`.